src/components/App/index.tsx

```diff
@@ -0,0 +1,294 @@
+    [versionLists, patchVersionsData]
+  );
+
+  // display popup when first visiting the site
+  useEffect(() => {
+    const cookieKey = 'visited-merge-notice';
+    if (!Cookies.get(cookieKey)) {
+      swal({
+        button: 'Got It!',
+        content: (
+          <div>
+            <img
+              style={{ width: '175px' }}
+              alt="GT Scheduler Logo"
+              src="/mascot.png"
+            />
+            <h1>GT Scheduler</h1>
+            <p>
+              Hey there, yellow jackets!{' '}
+              <a href="https://github.com/gt-scheduler">GT Scheduler</a> is a
+              new collaboration between{' '}
+              <a href="https://bitsofgood.org/">Bits of Good</a> and{' '}
+              <a href="https://jasonpark.me/">Jason (Jinseo) Park</a> aimed at
+              making class registration easier for everybody! Now, you can
+              access course prerequisites, instructor GPAs, live seating
+              information, and more all in one location.
+            </p>
+            <p>
+              If you enjoy our work and are interested in contributing, feel
+              free to{' '}
+              <a href="https://github.com/gt-scheduler/website/pulls">
+                open a pull request
+              </a>{' '}
+              with your improvements. Thank you and enjoy!
+            </p>
+          </div>
+        )
+      });
+
+      Cookies.set(cookieKey, 'true', { expires: 365 });
+    }
+  }, []);
+
+  // Fetch the current term's scraper information
+  useEffect(() => {
+    setOscar(null);
+    if (term) {
+      axios
+        .get(`https://gt-scheduler.github.io/crawler/${term}.json`)
+        .then((res) => {
+          const newOscar = new Oscar(res.data);
+          setOscar(newOscar);
+        });
+    }
+  }, [term]);
+
+  // Fetch all terms via the GitHub API
+  useEffect(() => {
+    axios
+      .get(
+        'https://api.github.com/repos/gt-scheduler/crawler/contents?ref=gh-pages'
+      )
+      .then((res) => {
+        const newTerms = (res.data as { name: string }[])
+          .map((content) => content.name)
+          .filter((name: string) => /\d{6}\.json/.test(name))
+          .map((name: string) => name.replace(/\.json$/, ''))
+          .sort()
+          .reverse();
+        setTerms(newTerms);
+      });
+  }, [setTerms]);
+
+  // Set the term to be the first one if it is unset
+  // (once the terms load)
+  useEffect(() => {
+    const t = Cookies.get('term');
+    if (!term && (!t || t === 'undefined')) {
+      const [recentTerm] = terms;
+      setTerm(recentTerm);
+    } else if (t && t !== 'undefined') {
+      setTerm(t);
+    }
+  }, [terms, term, setTerm]);
+
+  // Initialize the versionName to Primary
+  useEffect(() => {
+    const v = Cookies.get('version');
+    if (!versionName) {
+      setVersionName(v || 'Primary');
+    }
+  }, [versionName, setVersionName]);
+
+  // Initialize the version lists for each term
+  useEffect(() => {
```

Comment:
How will this handle when a new term is added to the crawler's data? Will the versions data get an additional element added in that case (either explicitly or implicitly?)

Reply:
This function initializes version lists based on the "terms" state, which is set based on get response from the crawler. Therefore, the initialization will add new lists explicitly when user first use the app. However, if the versions is already set, meaning the user already used the app and a new term is added, current implementation will not add a new list. I added new code to account for that; thanks for catching that!